packages/dev/core/src/FlowGraph/Blocks/Data/flowGraphGetVariableBlock.ts

```diff
@@ -17,10 +20,13 @@ export class FlowGraphGetVariableBlock<T> extends FlowGraphBlock {
      */
     public readonly output: FlowGraphDataConnection<T>;
 
-    constructor() {
+    constructor(params?: IFlowGraphGetVariableBlockParameter) {
         super();
 
         this.variableName = this._registerDataInput("variableName", RichTypeString);
+        if (params?.variableName) {
+            this.variableName.value = params.variableName;

```

Comment:
wouldn t the block be equivalent to: `this.variableName.value = params?.variableName;` ? I might be wrong here

Reply:
No because RichTypeString already initializes the input with a value (currently ""), so we don't want to overwrite it with undefined from params?.variableName